platforms/documentation/docs/src/docs/userguide/reference/best-practices/best_practices_tasks.adoc

```diff
@@ -409,3 +409,60 @@ The output reveals that the `lib` project is now built when the `:app:goodClassp
 
 === Tags
 `<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:inputs-and-outputs,#inputs-and-outputs>>`, `<<tags_reference.adoc#tag:configurations,#configurations>>`
+
+[[dont_hardcode_task_names]]
+== Don’t Hardcode Task Names unless they are Documented as Public API
+
+Hardcoding task names can make a build fragile when upgrading Gradle or third-party plugins.
+
+=== Explanation
+
+Some task names in Gradle are part of the <<public_apis.adoc#public_gradle_apis,Gradle public API>> and are stable.
+Most other task names in Gradle and third-party plugins are internal implementation details. They may be renamed or removed in future versions and should not be relied upon.
+Task types are also not guaranteed to remain stable; plugin authors may refactor them. 
+
+In practice, configuring tasks by type is usually more robust than depending on specific names, especially in ecosystems like Android or Kotlin where many tasks are generated dynamically.
+
+Prefer, in order:
+
+1. *Plugin DSL* - Future-proof and most stable.
+2. *Task types* - Robust in many cases, but can change across plugin versions.
+3. *Task names* - Use only when they are explicitly documented as <<public_apis.adoc#public_gradle_apis,public API>>.
+
+As a plugin author, provide a DSL extension for users to configure behavior declaratively. Internally wire the configuration into your tasks. Avoid exposing task names as part of your public API unless you are prepared to maintain them with deprecation cycles.
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/avoidHardcodingTaskNames-avoid/kotlin",files="build.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/avoidHardcodingTaskNames-avoid/groovy",files="build.gradle[tags=avoid-this]"]
+====
+
+<1> Using `tasks.named` to get `GenerateMavenPom`: This relies on this task name and that's the only `GenerateMavenPom` task that needs configuring.
+
+==== Do This Instead
+
+The following is okay to use because this task name is part of <<public_apis.adoc#public_gradle_apis,public API>>.
+
+====
+include::sample[dir="snippets/bestPractices/avoidHardcodingTaskNames-do/kotlin",files="build.gradle.kts[tags=ok]"]
+include::sample[dir="snippets/bestPractices/avoidHardcodingTaskNames-do/groovy",files="build.gradle[tags=ok]"]
+====
+
+<1> Using `tasks.named` to get `JavaCompile`: This uses a public constant to get `JavaCompile`.
```

Comment:
🤔 I think we should show configuring `javaCompile` in the "avoid" example as well, so that this isn't a huge difference between the avoid and do examples.

Perhaps "avoid" could show using "javaCompile" instead of the constant.  Then in a single "do" snippet, we could show both using the constant and the compatibility DSL.  Looking up this task wouldn't be a case of avoid/do, but rather of ok (inline String)/better (using the constant)/best (DSL).

Even if we don't do that, I think it's kind of jarring to go from reading the avoid example about maven publication to the first recommendation being about java compile when java compile wasn't even present in the avoid.